prow/config/jobs.go

```diff
@@ -415,6 +415,9 @@ func NewGitHubDeferredChangedFilesProvider(client githubClient, org, repo string
 type UtilityConfig struct {
 	// Decorate determines if we decorate the PodSpec or not
 	Decorate bool `json:"decorate,omitempty"`
+	// Undecorated determines if the job should not be decorated. Overrides
+	// other decorate options.
+	Undecorated bool `json:"undecorated,omitempty"`
```

Comment:
`Decorate *bool` `json:"decorate,omitempty"` is generally how we do this, which makes the behavior clear:

```yaml
# will decorate
decorate: true
---
# will not decorate
decorate: false
---
# will use default behavior
decorate: null # equivalent to it being missing
```

This seems unnecessarily messy -- aka what happens for `decorate: true, undecorated: true` jobs?

Reply:
A couple examples:

https://github.com/kubernetes/test-infra/blob/0110928adec52dd109255b9ebec3b7a8a1c89197/prow/config/jobs.go#L99-L102


https://github.com/kubernetes/test-infra/blob/0110928adec52dd109255b9ebec3b7a8a1c89197/prow/config/branch_protection.go#L31-L32

Reply:
Thank you for the review. Fixed.